catalog/files/impl/src/test/java/org/projectnessie/catalog/files/gcs/TestGcsProgrammaticOptions.java

```diff
@@ -0,0 +1,159 @@
+                    .readChunkSize(7)
+                    .writeChunkSize(8)
+                    .deleteBatchSize(9)
+                    .build())
+            .build();
+    GcsOptions expected =
+        ImmutableGcsProgrammaticOptions.builder()
+            .readTimeout(Duration.ofSeconds(1))
+            .connectTimeout(Duration.ofSeconds(2))
+            .maxAttempts(2)
+            .logicalTimeout(Duration.ofSeconds(3))
+            .totalTimeout(Duration.ofSeconds(4))
+            .initialRetryDelay(Duration.ofSeconds(5))
+            .maxRetryDelay(Duration.ofSeconds(6))
+            .retryDelayMultiplier(7)
+            .initialRpcTimeout(Duration.ofSeconds(8))
+            .maxRpcTimeout(Duration.ofSeconds(9))
+            .rpcTimeoutMultiplier(10)
+            .defaultOptions(
+                ImmutableGcsNamedBucketOptions.builder()
+                    .host(URI.create("https://host"))
+                    .externalHost(URI.create("https://externalHost"))
+                    .userProject("userProject")
+                    .projectId("projectId")
+                    .quotaProjectId("quotaProjectId")
+                    .clientLibToken("client lib token")
+                    .authType(GcsAuthType.NONE)
+                    .readChunkSize(1)
+                    .writeChunkSize(2)
+                    .deleteBatchSize(3)
+                    .build())
+            .putBuckets(
+                "bucket1",
+                ImmutableGcsNamedBucketOptions.builder()
+                    .name("bucket1")
+                    .host(URI.create("https://host1"))
+                    .externalHost(URI.create("https://externalHost1"))
+                    .userProject("userProject1")
+                    .projectId("projectId1")
+                    .quotaProjectId("quotaProjectId1")
+                    .clientLibToken("client lib token1")
+                    .authType(GcsAuthType.ACCESS_TOKEN)
+                    .readChunkSize(4)
+                    .writeChunkSize(5)
+                    .deleteBatchSize(6)
+                    .build())
+            .putBuckets(
+                "my-bucket-2",
+                ImmutableGcsNamedBucketOptions.builder()
+                    .name("my-bucket-2")
+                    .host(URI.create("https://host2"))
+                    .externalHost(URI.create("https://externalHost2"))
+                    .userProject("userProject2")
+                    .projectId("projectId2")
+                    .quotaProjectId("quotaProjectId2")
+                    .clientLibToken("client lib token2")
+                    .authType(GcsAuthType.SERVICE_ACCOUNT)
+                    .readChunkSize(7)
+                    .writeChunkSize(8)
+                    .deleteBatchSize(9)
+                    .build())
+            .build();
+    GcsProgrammaticOptions actual = GcsProgrammaticOptions.copyOf(input);
+    assertThat(actual).isEqualTo(expected);
+  }
+
+  @Test
+  void normalizeBuckets() {
```

Comment:
Would be good to test the good case as well, and that the bucket options are contained using the `.name()` in S3/GCS/ADLSOptions.